--- a/CMakeLists.txt	2021-06-30 05:19:55.000000000 -0300
+++ b/CMakeLists.txt	2021-07-04 12:11:18.407428048 -0300
@@ -100,6 +100,7 @@
 option(TBB4PY_BUILD "Enable tbb4py build" OFF)
 option(TBB_CPF "Enable preview features of the library" OFF)
 option(TBB_FIND_PACKAGE "Enable search for external oneTBB using find_package instead of build from sources" OFF)
+option(TBBBIND_BUILD "Enable tbbbind build" ON)
 
 if (NOT DEFINED BUILD_SHARED_LIBS)
     set(BUILD_SHARED_LIBS ON)
@@ -196,8 +197,10 @@
         add_subdirectory(src/tbbmalloc_proxy)
         if (APPLE)
             message(STATUS "TBBBind build target is disabled due to unsupported environment")
-        else()
+        elseif(TBBBIND_BUILD)
             add_subdirectory(src/tbbbind)
+        else()
+            message(STATUS "TBBBind build target is disabled")
         endif()
     endif()
 
