diff -urN tbb-2021.4.0/src/tbb/tools_api/ittnotify_config.h tbb-2021.4.0-fixed/src/tbb/tools_api/ittnotify_config.h
--- tbb-2021.4.0/src/tbb/tools_api/ittnotify_config.h	2021-10-04 03:50:18.000000000 -0600
+++ tbb-2021.4.0-fixed/src/tbb/tools_api/ittnotify_config.h	2021-10-04 18:23:19.852545545 -0600
@@ -159,6 +159,10 @@
 #  define ITT_ARCH_ARM64  6
 #endif /* ITT_ARCH_ARM64 */
 
+#ifndef ITT_ARCH_RISCV64
+#  define ITT_ARCH_RISCV64  15 // Leave some room so the patch doesn't have to be updated soon.
+#endif /* ITT_ARCH_RISCV64 */
+
 #ifndef ITT_ARCH
 #  if defined _M_IX86 || defined __i386__
 #    define ITT_ARCH ITT_ARCH_IA32
@@ -172,6 +176,10 @@
 #    define ITT_ARCH ITT_ARCH_ARM64
 #  elif defined __powerpc64__
 #    define ITT_ARCH ITT_ARCH_PPC64
+#  elif defined __riscv || defined __riscv__
+#    if defined __riscv_xlen && __riscv_xlen == 64
+#      define ITT_ARCH ITT_ARCH_RISCV64
+#    endif
 #  endif
 #endif
 
