diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ff2978..43e545c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -288,7 +288,7 @@ if(CMAKE_BUILD_TYPE MATCHES Debug OR CMAKE_BUILD_TYPE MATCHES DebugFull)
 else ()
     message("-- Building in Release mode")
     set ( DEBUG_BUILD 0 )
-    set(CMAKE_INSTALL_DO_STRIP TRUE)
+    set(CMAKE_INSTALL_DO_STRIP FALSE)
     message("-- CMAKE_INSTALL_DO_STRIP is " ${CMAKE_INSTALL_DO_STRIP})
     if(UNIX AND NOT APPLE)
         ## Avoid hardening-no-relro wrarning messages from lintian
@@ -301,7 +301,6 @@ else ()
             add_definitions(${EXTRA_CPPFLAGS})
             set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${EXTRA_CFLAGS}")
             set(LINKER_OPTIONS "${EXTRA_LDFLAGS}")
-            list(APPEND LINKER_OPTIONS "-s") ## strip binaries
         endif()
     endif()
     
@@ -322,9 +321,6 @@ else ()
     message("-- Shared Objects will be written into ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
 
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O2")        ## Optimize
-    if(CMAKE_COMPILER_IS_GNUCXX)
-        set(CMAKE_EXE_LINKER_FLAGS "-s")  ## Strip binary
-    endif(CMAKE_COMPILER_IS_GNUCXX)
 
     ## In release, add NDEBUG unless explicitly told not to
     if (NOT PREVENT_WX_ASSERTS MATCHES 0)
